test/v5/createForeignPropertyPeriodCumulativeSummary/CreateForeignPropertyPeriodCumulativeSummaryControllerSpec.scala

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v5.createForeignPropertyPeriodCumulativeSummary
+
+import api.controllers.{ControllerBaseSpec, ControllerTestRunner}
+import api.models.audit.{AuditEvent, AuditResponse, GenericAuditDetail}
+import api.models.domain.{BusinessId, Nino, TaxYear}
+import api.models.errors._
+import api.models.outcomes.ResponseWrapper
+import api.services.{MockAuditService, MockEnrolmentsAuthService, MockMtdIdLookupService}
+import config.MockAppConfig
+import play.api.Configuration
+import play.api.libs.json.{JsObject, JsValue}
+import play.api.mvc.Result
+import utils.MockIdGenerator
+import v5.createForeignPropertyPeriodCumulativeSummary.def1.model.Def1_CreateForeignPropertyPeriodCumulativeSummaryFixtures
+import v5.createForeignPropertyPeriodCumulativeSummary.def1.model.request.Def1_CreateForeignPropertyPeriodCumulativeSummaryRequestData
+import v5.createForeignPropertyPeriodCumulativeSummary.model.request.CreateForeignPropertyPeriodCumulativeSummaryRequestData
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class CreateForeignPropertyPeriodCumulativeSummaryControllerSpec
+    extends ControllerBaseSpec
+    with MockAppConfig
+    with ControllerTestRunner
+    with MockEnrolmentsAuthService
+    with MockMtdIdLookupService
+    with MockCreateForeignPropertyPeriodCumulativeSummaryService
+    with MockCreateForeignPropertyPeriodCumulativeSummaryValidatorFactory
+    with MockAuditService
+    with MockIdGenerator
+    with Def1_CreateForeignPropertyPeriodCumulativeSummaryFixtures {
+
+  private val taxYear    = "2025-26"
+  private val businessId = "XAIS12345678910"
+
+  "CreateForeignPropertyPeriodCumulativeSummaryControllerSpec" should {
+    "return a successful response with status 204 (NO_CONTENT)" when {
+      "the request received is valid" in new Test {
+        willUseValidator(returningSuccess(requestData))
+
+        MockedCreateForeignPropertyPeriodCumulativeSummaryService
+          .createForeignProperty(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, ()))))
+
+        runOkTest(expectedStatus = NO_CONTENT, maybeExpectedResponseBody = None)
+
+      }
+
+      "return the error as per spec" when {
+        "the parser validation fails" in new Test {
+          willUseValidator(returning(NinoFormatError))
+
+          runErrorTest(NinoFormatError)
+        }
+
+        "the service returns an error" in new Test {
+          willUseValidator(returningSuccess(requestData))
+
+          MockedCreateForeignPropertyPeriodCumulativeSummaryService
+            .createForeignProperty(requestData)
+            .returns(Future.successful(Left(ErrorWrapper(correlationId, RuleMisalignedPeriodError))))
```

Comment:
Double check this is the same error as is needed for CS (check if mapped correctly)